From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dofs <157043656+dofsdev@users.noreply.github.com>
Date: Thu, 7 Aug 2025 00:27:59 +0100
Subject: [PATCH] PATCH allowing charging of trident while not in water or rain


diff --git a/net/minecraft/world/item/TridentItem.java b/net/minecraft/world/item/TridentItem.java
index fac4c58ea5d467a8686e42676e2323fbddeb8c7b..4d23b572359e112610e2ad9e824b057a558c7fe2 100644
--- a/net/minecraft/world/item/TridentItem.java
+++ b/net/minecraft/world/item/TridentItem.java
@@ -149,8 +149,6 @@ public class TridentItem extends Item implements ProjectileItem {
         ItemStack itemInHand = player.getItemInHand(hand);
         if (itemInHand.nextDamageWillBreak()) {
             return InteractionResult.FAIL;
-        } else if (EnchantmentHelper.getTridentSpinAttackStrength(itemInHand, player) > 0.0F && !player.isInWaterOrRain()) {
-            return InteractionResult.FAIL;
         } else {
             player.startUsingItem(hand);
             return InteractionResult.CONSUME;
